### PR TITLE
[Moments] - Add cid field and patch request

### PR DESCRIPTION
**Pull request by nicolasdeleon**
## Description

Adding the cid field to the get Moments calls, that reflects the moment stored in Registry as well as the new patch Moment method that currently only supports cid and that has auth0 protection.

The cid of the moment in Registry under the format: `<Version>-<Signer>-<Signature>.<Type>` allows you to fetch and verify the content of the moment in Registry.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/poap-xyz/poap.js/blob/main/documentation/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by jm42 — COMMENTED** [review]
Better use `fetch` instead of `axios` when progress is not needed.

**Review by nacho9900 — APPROVED** [review]

**Review by Frannx1 — APPROVED** [review]

File: packages/providers/src/ports/MomentsApiProvider/types/CreateMomentResponse.ts [hunk 1]
```diff
@@ -25,4 +25,9 @@ export interface CreateMomentResponse {
    * The description of the moment.
    */
   description?: string;
+
+  /**
+   * The description of the moment.
```
Comment (rlajous) [inline packages/providers/src/ports/MomentsApiProvider/types/CreateMomentResponse.ts:30]: This is not the description of the moment

File: packages/providers/src/ports/MomentsApiProvider/types/PatchMomentInput.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Object describing the input required to update a Moment.
+ * @typedef {Object} PatchMomentInput
+ * @property {string} cid - The cid of the moment in Registry
```
Comment (rlajous) [inline packages/providers/src/ports/MomentsApiProvider/types/PatchMomentInput.ts:4]: What's a CID?

File: examples/moments/backend/src/methods/patch_moment.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { MomentsClient, PatchMomentInput } from '@poap-xyz/moments';
+
+export const patch_moment = async (client: MomentsClient): Promise<void> => {
+  const momentId = "b08fad41-7154-499f-88f9-abea66ceab48"
+  const input: PatchMomentInput = {
+    cid: "0001-7ce5368171cc3d988157d7dab3d313d7bd43de3e-365e5b83699adce0825021d011f1bf73bd5ef9369d06e49645afbea2ef34f54e0557c1d4742c8bd6d1f7a02be4aa483c03888af0aa143d5aa7351e2baaf931231c.moment",
```
Comment (rlajous) [inline examples/moments/backend/src/methods/patch_moment.ts:6]: Is there a way we can explain how this CID is generated? Where to get it?

File: packages/providers/src/core/PoapMomentsApi/PoapMomentsApi.ts [hunk 4]
```diff
@@ -164,6 +165,22 @@ export class PoapMomentsApi implements MomentsApiProvider {
     }
   }
 
+  /**
+   * Update a moment using the provided input
+   * @param {string} id - The moment id
+   * @param {PatchMomentInput} input - The input for updating a moment
+   */
+  public async patchMoment(id: string, input: PatchMomentInput): Promise<void> {
```
Comment (rlajous) [inline packages/providers/src/core/PoapMomentsApi/PoapMomentsApi.ts:173]: What errors can this endpoint generate?

**Review by rlajous — CHANGES_REQUESTED** [review]
It will be a good moment to update the documentation.
`docs/pages/packages/moments.mdx`

Also, you should have updated the library version.

File: examples/moments/backend/src/methods/patch_moment.ts (hunk 3 above)
Comment (nicolasdeleon) [reply examples/moments/backend/src/methods/patch_moment.ts:6]: We can address this in the moments docs with a detailed explanation of how Registry, CIDs and moments work together

File: packages/providers/src/core/PoapMomentsApi/PoapMomentsApi.ts (hunk 4 above)
Comment (nicolasdeleon) [reply packages/providers/src/core/PoapMomentsApi/PoapMomentsApi.ts:173]: ATM 404 moment not found and BadPatchRequest when cid is invalid but we are not doing anything with those errors. We can see of adding a good error handling strategy when this becomes an issue if we are ok

File: examples/moments/backend/src/methods/patch_moment.ts (hunk 3 above)
Comment (nicolasdeleon) [reply examples/moments/backend/src/methods/patch_moment.ts:6]: https://github.com/poap-xyz/poap.js/issues/118 will add this here

**Review by rlajous — APPROVED** [review]

**Comment by rlajous** [comment]
Could you let me know if you created a test version to see if it worked? 

- [ ] Mark this if done.

**Comment by nicolasdeleon** [comment]
> Could you let me know if you created a test version to see if it worked?
> 
> * [x]  Mark this if done.

Here: https://www.npmjs.com/package/@poap-xyz/providers/v/0.2.7-beta-v1
And test it here: https://moments-frontend-12p23kzc2-poap-core.vercel.app/

**Pull request merged**